### PR TITLE
Fix printings of attributes on Ptyp_arrow

### PR DESCRIPTION
**Pull request by IwanKaramazow**
Fixes https://github.com/facebook/reason/issues/837

`ptyp_attributes` on `core_type`'s where ignored at some point in the code when formatting `Ptyp_arrow`'s. It resulted in attributes disappearing. The solution is to format `Ptyp_arrow`'s with attributes as a separate core_type instead of recursing through it's tree.

```reason
/* input */
external debounce : (int, [@bs.meth] unit) => unit = "";
external debounce : int => ([@bs.meth] (unit => unit)) = "";
external debounce : (int, ([@bs.meth] (unit => unit))) => ([@bs.meth] (unit => unit)) = "";

/* after refmt */
external debounce : (int, [@bs.meth] unit) => unit = "";
external debounce : int => [@bs.meth] (unit => unit) = "";
external debounce : (int, [@bs.meth] (unit => unit)) => [@bs.meth] (unit => unit) = "";
```

### CODE REVIEW

**Comment by chenglou** [comment]
Humm this doesn't seem right:

```sh
❯ echo 'external debounce : int => ([@bs.meth] (unit => unit)) = "";' | ./refmt_impl.native --parse re --print ml
external debounce : int -> ((unit -> unit)[@bs.meth ]) = ""
```

```sh
❯ echo 'external debounce : int => ([@bs.meth] (unit => unit)) = "";' | ./refmt_impl.native --parse re --print re
external debounce : (int, [@bs.meth] unit) => unit = "";
```

**Comment by IwanKaramazow** [comment]
Sorry, should have tested more thorough.
Pushed the fix, which seems to be a one-liner...

```bash
echo 'external debounce : int => ([@bs.meth] (unit => unit)) = "";' | ./refmt_impl.native --parse re --print ml
external debounce : int -> ((unit -> unit)[@bs.meth ]) = ""
```

```bash
echo 'external debounce : int => ([@bs.meth] (unit => unit)) = "";' | ./refmt_impl.native --parse re --print re
external debounce : int => [@bs.meth] (unit => unit) = "";
```

**Comment by jordwalke** [comment]
Awesome, thank you @IwanKaramazow !

File: src/reason_pprint_ast.ml [hunk 1]
```diff
@@ -2561,16 +2561,17 @@ class printer  ()= object(self:'self)
       match (x.ptyp_desc) with
         | (Ptyp_arrow (l, ct1, ct2)) ->
           let rec allArrowSegments acc = function
-            | { ptyp_desc = Ptyp_arrow (l, ct1, ct2); _ } ->
+            | { ptyp_desc = Ptyp_arrow (l, ct1, ct2); ptyp_attributes } when List.length ptyp_attributes == 0 ->
```
Comment (chenglou) [inline src/reason_pprint_ast.ml]: Would this be better? `ptyp_attributes = []`

Comment (IwanKaramazow) [reply src/reason_pprint_ast.ml]: updated

**Comment by IwanKaramazow** [comment]
I have no idea why travis hanged...
Circleci seems to be ok though.

**Comment by chenglou** [comment]
Looks great. Merged! Thanks, as always

**Pull request merged**